src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement } from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+import { filterNodeByType } from './filterNodeByType';
+
+export const assertStringsWithinText = (instance: ReactTestInstance): void => {
+  const nodesWithStringChild = instance.findAll(hasNodeStringChild);
+  nodesWithStringChild.forEach((node) => {
+    const isHostTextComponent = filterNodeByType(node, 'Text');
+    const isNativeComponent = typeof node.type === 'string';
+
+    if (!isHostTextComponent && isNativeComponent) {
+      throw new Error(
+        'Text strings must be rendered within a host Text component.'
+      );
+    }
+  });
+};
+
+const hasNodeStringChild = (node: ReactTestInstance) => {
```

Comment:
The logic here is kinda permissive, as it allows to have a composite component, e.g. function/class component, that returns `string` value. That component would probably pass this check.

Reply:
I wonder if we would be even able to make check, as there as problems discussed in #967, which you probably can remember @pierrezimmermannbam ;-)